botocore/operation.py

```diff
@@ -86,8 +113,25 @@ def call(self, endpoint, **kwargs):
         self.session.emit(event, endpoint=endpoint,
                           model=self.model,
                           params=request_dict,
-                          operation=self)
+                          operation=self,
+                          request_signer=signer)
+
+        # This is kind of ugly, but it lets us hook into the request
+        # created event on a per-operation basis without mucking with
+        # the Endpoint/EndpointCreator objects.
+        real_emit = event_emitter.emit
+        def fake_emit(name, **kwargs):
```

Comment:
Not a fan of this.  This has a number of problems, and is also not thread safe.  Also, as a best practice, we should not be monkey patching code like this.  Let's come up with a cleaner way to do this.